### PR TITLE
[data wrapping service] enable minio server-side encryption

### PR DESCRIPTION
**Pull request by matthewrossi**
This PR relates to https://github.com/glaciation-heu/IceStream/issues/130 and depends on https://github.com/glaciation-heu/gitops-deployments/pull/44.

It updates the MinIO tenant configuration to enable server-side encryption with the integration of HaschiCorp Vault for the key management. Since the MinIO tenant depends on the deployment of the MinIO operator, here we have also anticipated the deployment of the MinIO operator to the same ArgoCD wave of the MinIO tenant.

### CODE REVIEW

File: integration/apps/minio-tenant.yaml [hunk 1]
```diff
@@ -1,536 +1,202 @@
-apiVersion: argoproj.io/v1alpha1
-kind: Application
+apiVersion: v1
+kind: Secret
 metadata:
-  name: minio-tenant
-  namespace: argocd
+  name: glaciation-env-configuration
```
Comment (ktatarnikovhiro) [inline integration/apps/minio-tenant.yaml:4]: I am looking and see that this minio-tenant is very different from the one deployed from helm chart. Is there a way to deploy it from helm chart? if there is no way then I propose to make a separate tenant for data wrapping.

File: integration/apps/minio-tenant.yaml [hunk 2]
```diff
@@ -1,536 +1,202 @@
...
+---
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: minio-tenant-api
+  namespace: minio-tenant
+  annotations:
+    # NOTE: To make sure the certificate is trusted by clients we should use
+    # something like Let's Encrypt
+    cert-manager.io/cluster-issuer: "private-ca-issuer"
```
Comment (ktatarnikovhiro) [inline integration/apps/minio-tenant.yaml:154]: I am afraid TLS is not available in the integration cluster in ingresses and ssl-redirect as well.

**Review by ktatarnikovhiro — CHANGES_REQUESTED** [review]

File: integration/apps/minio-tenant.yaml (hunk 1 above)
Comment (matthewrossi) [reply integration/apps/minio-tenant.yaml]: I managed to make it work with the Helm chart

File: integration/apps/minio-tenant.yaml (hunk 2 above)
Comment (matthewrossi) [reply integration/apps/minio-tenant.yaml]: We could do TLS termination at ingress (like above), and then export the service without TLS, but I would suggest to go for this solution only after checking the current ingress does not work with the integration cluster.

Comment (ktatarnikovhiro) [reply integration/apps/minio-tenant.yaml]: I am merging it and we will see...

**Pull request merged**